### PR TITLE
Add instruction to install DTK on Summitdev.

### PR DESCRIPTION
**Pull request by Rombur**
Not sure if we should to turn `DBC` ON or not.

### CODE REVIEW

File: scripts/README_OLCF_SUMMITDEV.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Building DTK on Summitdev
+The user guide for Summitdev can be found [here](https://www.olcf.ornl.gov/kb_articles/summitdev-quickstart).
+
+## Software environment
+Below are the modules that need to be loaded:
+```
+module swap xl gcc/5.4.0
```
Comment (aprokop) [inline scripts/README_OLCF_SUMMITDEV.md]: So, no building with IBM compiler?

Comment (Rombur) [inline scripts/README_OLCF_SUMMITDEV.md]: didn't try

File: scripts/README_OLCF_SUMMITDEV.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+module load cmake
+module load cuda
+module load boost
+module load netlib-lapack
+```
+
+# Configuring Trilinos with DTK
+This directory constains a configuration script (olcf_summitdev_cmake) to build
+DTK and its test suite. Don't forget to set `CMAKE_INSTALL_PREFIX` and
+`TRILINOS_DIR`. You also need to set `nvcc` as the underlying C++ compiler for
```
Comment (aprokop) [inline scripts/README_OLCF_SUMMITDEV.md]: you mean `nvcc_wrapper` here?

**Comment by dalg24** [comment]
why not a single README with multiple sections?

File: scripts/README_OLCF_SUMMITDEV.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+Spectrum MPI:
```
Comment (aprokop) [inline scripts/README_OLCF_SUMMITDEV.md]: Spectrum?

File: scripts/README_OLCF_SUMMITDEV.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+```
+export OMPI_CXX=$TRILINOS_DIR/packages/kokkos/config/nvcc_wrapper
+```
+
+# CUDA-aware MPI
+By default, CUDA-awareness is disable by default. To enable it, add the `-gpu`
+flag, to `mpirun`.
```
Comment (aprokop) [inline scripts/README_OLCF_SUMMITDEV.md]: Is there an environment variable for this?

**Comment by aprokop** [comment]
Should the instructions go to RTD?

File: scripts/README_OLCF_SUMMITDEV.md (hunk 2 above)
Comment (Rombur) [inline scripts/README_OLCF_SUMMITDEV.md]: Well the wrapper put `nvcc` as the underlying C++ compiler.

File: scripts/README_OLCF_SUMMITDEV.md (hunk 3 above)
Comment (Rombur) [inline scripts/README_OLCF_SUMMITDEV.md]: IBM MPI is called Spectrum

File: scripts/README_OLCF_SUMMITDEV.md (hunk 4 above)
Comment (dalg24) [inline scripts/README_OLCF_SUMMITDEV.md]: How did you achieve that?  Did you make an alias?  Or did you pass `-D MPI_EXEC="mpirun -gpu"`?

Comment (Rombur) [inline scripts/README_OLCF_SUMMITDEV.md]: Don't think so. There is nothing in the documentation about using an environment variable.

Comment (Rombur) [inline scripts/README_OLCF_SUMMITDEV.md]: I don't understand. We don't use `mpi`...

File: scripts/olcf_summit_dev_cmake [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    -D TPL_ENABLE_MPI=ON
+    -D TPL_ENABLE_BLAS=ON
+    -D TPL_ENABLE_LAPACK=ON
+    -D TPL_ENABLE_CUDA=ON
+    -D TPL_ENABLE_Boost=ON
+        -D Boost_INCLUDE_DIRS=$BOOST_DIR/include
+        -D Boost_LIBRARY_DIRS=$BOOST_DIR/lib
+    -D TPL_ENABLE_BoostLib=ON
+        -D BoostLib_INCLUDE_DIRS=$BOOST_DIR/include
+        -D BoostLib_LIBRARY_DIRS=$BOOST_DIR/lib
```
Comment (dalg24) [inline scripts/olcf_summit_dev_cmake]: interesting that they set the env variable `BOOST_DIR` when CMake usually relies on `BOOST_ROOT`.  did TriBITS not detect Boost without you specifying the include and lib paths?

File: scripts/README_OLCF_SUMMITDEV.md (hunk 2 above)
Comment (aprokop) [inline scripts/README_OLCF_SUMMITDEV.md]: but you set `nvcc_wrapper` as the underlying compiler, which in turn calls `nvcc`.

File: scripts/olcf_summit_dev_cmake (hunk 5 above)
Comment (aprokop) [inline scripts/olcf_summit_dev_cmake]: Does CMake really provide a default name for an environment variable? Don't think I've seen that.

Comment (Rombur) [inline scripts/olcf_summit_dev_cmake]: Actually, I just copied from our configure file and it worked (I did check that the libraries that were picked up where correct).

**Comment by Rombur** [comment]
@dalg24 Personally, I find it easier to find what I need this way. We can have one big `README` if you prefer.

File: scripts/olcf_summit_dev_cmake (hunk 5 above)
Comment (dalg24) [inline scripts/olcf_summit_dev_cmake]: I cannot remember what TriBITS does with `<TPL>_{INCLUDE,LIB}_DIRS`, if does prepend it to the search path or if it throw an error if the path provided does not contains whatever it is looking for.  (I suspect the former.)

Comment (dalg24) [inline scripts/olcf_summit_dev_cmake]: is `BOOST_DIR` even defined in your environment after you load the module?

Comment (Rombur) [inline scripts/olcf_summit_dev_cmake]: yeah it looks like you are right. I will get ride of these lines

**Comment by dalg24** [comment]
>  Personally, I find it easier to find what I need this way. We can have one big README if you prefer.

Fair enough, but the doc on how to generate the changeling does not belong to `README_ALCF` and we would probably want a regular `README.md` that references both ALCF and OLCF instructions then.

File: scripts/README_OLCF_SUMMITDEV.md (hunk 2 above)
Comment (Rombur) [inline scripts/README_OLCF_SUMMITDEV.md]: so `nvcc` is the real underlying compiler...

File: scripts/README_OLCF_SUMMITDEV.md (hunk 4 above)
Comment (dalg24) [inline scripts/README_OLCF_SUMMITDEV.md]: > to enable it, add the `-gpu` flag, to `mpirun`.

**Comment by Rombur** [comment]
@dalg24 I have removed `BOOST_DIR` and added a global `README`

File: scripts/README_OLCF_SUMMITDEV.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+module load cmake
+module load cuda
+module load boost
+module load netlib-lapack
+```
+
+# Configuring Trilinos with DTK
+This directory constains a configuration script
+([olcf_summitdev_cmake](olcf_summitdev_cmake)) to build DTK and its test suite.
+Don't forget to set `CMAKE_INSTALL_PREFIX` and `TRILINOS_DIR`. You also need
```
Comment (dalg24) [inline scripts/README_OLCF_SUMMITDEV.md]: Also don't forget to create a symbolic link for DTK in it or to clone the repo directly into it ;)

File: scripts/README_OLCF_SUMMITDEV.md (hunk 2 above)
Comment (aprokop) [inline scripts/README_OLCF_SUMMITDEV.md]: Sorry for pushing it. We both understand how things work.  But to me when you say "you also need to set X as underlying compiler" it means "OMPI_CXX=X", which is not true in this case.

**Comment by Rombur** [comment]
@aprokop done

**Pull request merged**